File: JavaFundamentals/src/Methods/Exercise/PasswordValidator.java
package Methods.Exercise;

import java.util.Scanner;

public class PasswordValidator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String password = scanner.nextLine();
        boolean isValid = true;

        if (!isLengthValid(password)) {
            System.out.println("Password must be between 6 and 10 characters");
            isValid = false;
        }

        if (!containsOnlyLettersAndDigits(password)) {
            System.out.println("Password must consist only of letters and digits");
            isValid = false;
        }

        if (!hasAtLeastTwoDigits(password)) {
            System.out.println("Password must have at least 2 digits");
            isValid = false;
        }

        if (isValid) {
            System.out.println("Password is valid");
        }
    }

    private static boolean isLengthValid(String password) {
        return password.length() >= 6 && password.length() <= 10;
    }

    private static boolean containsOnlyLettersAndDigits(String password) {
        for (char c : password.toCharArray()) {
            if (!Character.isLetterOrDigit(c)) {
                return false;
            }
        }
        return true;
    }

    private static boolean hasAtLeastTwoDigits(String password) {
        int digitCount = 0;
        for (char c : password.toCharArray()) {
            if (Character.isDigit(c)) {
                digitCount++;
            }
        }
        return digitCount >= 2;
    }
}

File: JS Front End/Functions and Statements - Exercises/passwordValidator.js
function passwordValidator(password) {
    function isLetter(ch) {
        return ch.toLowerCase() !== ch.toUpperCase()
    }

    function isDigit(ch) {
        return !isNaN(ch)
    }

    const result = [];

    if (!(password.length >= 6 && password.length <= 10)) {
        result.push('Password must be between 6 and 10 characters')
    }

    let digits = 0;
    for (let char of password) {
        if (!(isLetter(char) || isDigit(char))) {
            result.push('Password must consist only of letters and digits');
            break;
        }

        if (isDigit(char)) {
            digits += 1;
        }
    }

    if (digits < 2) {
        result.push('Password must have at least 2 digits');
    }

    result.length > 0
        ? console.log(result.join('\n'))
        : console.log('Password is valid')
}

passwordValidator('logIn');
passwordValidator('MyPass123');
passwordValidator('Pa$s$s');